Sources/HummingbirdWebSocket/Client/WebSocketClient.swift

```diff
@@ -0,0 +1,273 @@
+                    logger: self.logger
+                )
+                try await client.run()
+
+            #endif
+            case .none:
+                let client = try HBClient(
+                    TLSClientChannel(
+                        WebSocketClientChannel(
+                            handler: handler,
+                            url: urlPath,
+                            maxFrameSize: self.configuration.maxFrameSize,
+                            additionalHeaders: self.configuration.additionalHeaders
+                        ),
+                        tlsConfiguration: TLSConfiguration.makeClientConfiguration()
+                    ),
+                    address: .hostname(host, port: port),
+                    eventLoopGroup: self.eventLoopGroup,
+                    logger: self.logger
+                )
+                try await client.run()
+            }
+        } else {
+            let client = HBClient(
+                WebSocketClientChannel(
+                    handler: handler,
+                    url: urlPath,
+                    maxFrameSize: self.configuration.maxFrameSize,
+                    additionalHeaders: self.configuration.additionalHeaders
+                ),
+                address: .hostname(host, port: port),
+                eventLoopGroup: self.eventLoopGroup,
+                logger: self.logger
+            )
+            try await client.run()
+        }
+    }
+}
+
+extension HBWebSocketClient {
+    /// Create websocket client, connect and handle connection
+    ///
+    /// - Parametes:
+    ///   - url: URL of websocket
+    ///   - tlsConfiguration: TLS configuration
+    ///   - maxFrameSize: Max frame size for a single packet
+    ///   - eventLoopGroup: EventLoopGroup to run WebSocket client on
+    ///   - logger: Logger
+    ///   - process: Closure handling webSocket
+    public static func connect(
+        url: HBURL,
+        configuration: Configuration = .init(),
+        tlsConfiguration: TLSConfiguration? = nil,
+        eventLoopGroup: EventLoopGroup = MultiThreadedEventLoopGroup.singleton,
+        logger: Logger,
+        process: @escaping HBWebSocketDataCallbackHandler.Callback
+    ) async throws {
+        let ws = try self.init(
+            url: url,
+            configuration: configuration,
+            tlsConfiguration: tlsConfiguration,
+            eventLoopGroup: eventLoopGroup,
+            logger: logger,
+            process: process
+        )
+        try await ws.run()
+    }
+
+    #if canImport(Network)
+    /// Create websocket client, connect and handle connection
+    ///
+    /// - Parametes:
+    ///   - url: URL of websocket
+    ///   - transportServicesTLSOptions: TLS options for NIOTransportServices
+    ///   - maxFrameSize: Max frame size for a single packet
+    ///   - eventLoopGroup: EventLoopGroup to run WebSocket client on
+    ///   - logger: Logger
+    ///   - process: WebSocket data handler
+    public static func connect(
```

Comment:
Would it be reasonable for users to want Network.framework but not TLS?

Reply:
They can use a NIOTSEventLoopGroup for that